crates/hir-ty/src/infer.rs

```diff
@@ -2266,123 +2212,3 @@ impl std::ops::BitOrAssign for Diverges {
         *self = *self | other;
     }
 }
-
-/// A zipper that checks for unequal occurrences of `{unknown}` and unresolved projections
-/// in the two types. Used to filter out mismatch diagnostics that only differ in
-/// `{unknown}` and unresolved projections. These mismatches are usually not helpful.
-/// As the cause is usually an underlying name resolution problem
-struct UnknownMismatch<'db>(&'db dyn HirDatabase);
```

Comment:
The reason I removed this is because the next solver, unlike Chalk, does not have a trait for zipping (it has `Relate` but it has some things that prevent it from being used for that easily, such as variance). I didn't want to introduce our own trait just for that, and also considering our nameres got considerably better during the years perhaps this is no longer needed.